src/modules/fundingManager/depositVault/FM_DepositVault_v1.sol

```diff
@@ -0,0 +1,130 @@
+    "@oz-up/utils/introspection/ERC165Upgradeable.sol";
+
+// External Libraries
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title   Deposit Vault Funding Manager
+ *
+ * @notice  This contract allows users to deposit tokens to fund the workflow.
+ *
+ * @dev     Implements {IFundingManager_v1} interface.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to our Security Policy
+ *                          at security.inverter.network or email us directly!
+ *
+ * @author  Inverter Network
+ */
+contract FM_DepositVault_v1 is
+    IFundingManager_v1,
+    IFM_DepositVault_v1,
+    Module_v1
+{
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override(Module_v1)
+        returns (bool)
+    {
+        return interfaceId == type(IFundingManager_v1).interfaceId
+            || interfaceId == type(IFM_DepositVault_v1).interfaceId
+            || super.supportsInterface(interfaceId);
+    }
+
+    using SafeERC20 for IERC20;
+
+    //--------------------------------------------------------------------------
+    // Storage
+
+    /// @dev The token that is deposited.
+    IERC20 private _token;
+
+    /// @dev Storage gap for future upgrades
+    uint[50] private __gap;
+
+    //--------------------------------------------------------------------------
+    // Init Function
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata,
+        bytes memory configData
+    ) external override initializer {
+        __Module_init(orchestrator_, metadata);
+
+        address orchestratorTokenAddress = abi.decode(configData, (address));
+        _token = IERC20(orchestratorTokenAddress);
+    }
+
+    //--------------------------------------------------------------------------
+    // Public View Functions
+
+    /// @inheritdoc IFundingManager_v1
+    function token() public view returns (IERC20) {
+        return _token;
+    }
+
+    //--------------------------------------------------------------------------
+    // Public Mutating Functions
+
+    /// @inheritdoc IFM_DepositVault_v1
+    function deposit(uint amount) external {
+        address from = _msgSender();
+        _deposit(from, amount);
+    }
+
+    /// @inheritdoc IFM_DepositVault_v1
+    function depositFor(address from, uint amount) external {
+        _deposit(from, amount);
+    }
+
+    //--------------------------------------------------------------------------
+    // Internal Functions
+
+    /// @dev Internal function to deposit tokens.
+    /// @param from The address to deposit tokens for.
+    /// @param amount The number of tokens to deposit.
+    function _deposit(address from, uint amount) internal {
+        token().safeTransferFrom(from, address(this), amount);
```

Comment:
So if I approved 20 tokens to this contract, but only wanted to deposit 10, anyone can call depositFor and make it deposit the other 10 tokens as well. This seems quite dangerous. A normal depositFor behaviour would be that the tokens are taken from the msgSender, but are credited to the from address.

Reply:
I'd say what is implemented here is more "depositFrom" than the usual "depositFor"

Reply:
Very valid point, this is not the traditional `depositFor` where a user would get receipt tokens, withdraw rights, or anything else in return. It's a bit confusing. I'd probably just rename it then to `depositFrom`, but...

I'd personally just remove this function. Looking at it now, I don't see any reason to have it. If someone can't (for whatever reason) call the funding manager directly to deposit (weird wallet that can't call arbitrary contracts, or whatever), they can always just transfer the tokens directly into the DepositVault anyway via a regular token transfer, as it doesn't matter who sent the tokens and no accounting happening.

I've removed it in [this commit](https://github.com/InverterNetwork/contracts/pull/646/commits/ef74cf6e825dc685712d95b1f9e3614da9c0eb20). The PR handling all fixes is #626. Let me know what you think @ben-kaufman - I'll also give @Zitzak and @0xNuggan the chance to comment here as well, maybe I missed something, in which case we can still revert my removal of course.

Reply:
Yes I think it makes sense to remove it